src/gluonts/ev/aggregations.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from typing import List, Optional, Union
+import numpy as np
+
+from gluonts.ev.helpers import axis_is_zero_or_none
+
+
+@dataclass
+class Aggregation:
+    def step(self) -> None:
+        raise NotImplementedError
+
+    def get(self) -> np.ndarray:
+        raise NotImplementedError
+
+    def reset(self) -> None:
+        raise NotImplementedError
+
+
+@dataclass
+class Sum(Aggregation):
+    axis: Optional[int] = None
+    partial_result: Optional[Union[List[np.ndarray], np.ndarray]] = None
+
+    def step(self, values) -> None:
+        summed_values = np.sum(values, axis=self.axis)
+
+        if self.partial_result is None:
```

Comment:
Maybe we can refactor this to make it more clear that is only invoked on the first execution.